File: aoj/introduction/IntroductIonToProgramming/1_6_B_Finding_Missing_Cards/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        boolean[][] cards = new boolean[4][14];

        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {

            char suit = sc.next().charAt(0);
            int rank = sc.nextInt();

            switch (suit) {
                case 'S':
                    cards[0][rank] = true;
                    break;
                case 'H':
                    cards[1][rank] = true;
                    break;
                case 'C':
                    cards[2][rank] = true;
                    break;
                case 'D':
                    cards[3][rank] = true;
                    break;
            }

        }

        for (int i = 0; i < cards.length; i++) {

            for (int j = 1; j < cards[i].length; j++) {

                if (cards[i][j] == false) {

                    switch (i) {
                        case 0:
                            System.out.println("S " + j);
                            break;
                        case 1:
                            System.out.println("H " + j);
                            break;
                        case 2:
                            System.out.println("C " + j);
                            break;
                        case 3:
                            System.out.println("D " + j);
                            break;
                    }

                }

            }

        }

    }

}

File: aoj/introduction/IntroductIonToProgramming/1_6_B_Finding_Missing_Cards/Main.cpp
#include <iostream>

using namespace std;

int main() {

    bool cards[4][14] = {{0}};

    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {

        char suit;
        int rank;
        cin >> suit >> rank;

        switch (suit) {
            case 'S':
                cards[0][rank] = true;
                break;
            case 'H':
                cards[1][rank] = true;
                break;
            case 'C':
                cards[2][rank] = true;
                break;
            case 'D':
                cards[3][rank] = true;
                break;
        }

    }

    for (int i = 0; i < 4; i++) {
        for (int j = 1; j < 14; j++) {

            if (cards[i][j] == false) {

                switch (i) {
                    case 0:
                        cout << "S " << j << endl;
                        break;
                    case 1:
                        cout << "H " << j << endl;
                        break;
                    case 2:
                        cout << "C " << j << endl;
                        break;
                    case 3:
                        cout << "D " << j << endl;
                        break;
                }

            }

        }
    }

    return 0;

}
